typing/env.ml

```diff
@@ -3655,8 +3664,45 @@ let report_lookup_error_doc _loc env ppf = function
         "The ancestor variable %a@ \
          cannot be accessed from the definition of an instance variable"
        quoted_longident lid
-  | Illegal_reference_to_recursive_module ->
-     fprintf ppf "Illegal recursive module reference"
+  | Illegal_reference_to_recursive_module { container; unbound } ->
+      let container = Option.value ~default:"_" container in
+      fprintf ppf
+        "@[<hov>This module type is recursive.@ \
+         This use of the recursive module %a@ \
+         within the definition of the module %a@ \
+         makes the module type of %a depend on "
+        Style.inline_code unbound
+        Style.inline_code container
+        Style.inline_code container;
+      if String.equal container unbound
+      then fprintf ppf "itself"
+      else fprintf ppf "the module type of %a" Style.inline_code unbound;
```

Comment:
Nitpick: I would rather use `%t` here than split the formatter in several commands printing parts of a sentence.

Reply:
Oh nice, I didn't even know about `%t`. Done.